src/rtc/mosaic_geobursts.py

```diff
@@ -0,0 +1,321 @@
+                flag_rtn = True
+
+            elif spacing_y_prev != geo_transformation_nlooks[5]:
+                print(f'spacing_y anomaly detected from : {os.path.basename(path_nlooks)}')
+                flag_rtn = True
+
+            # snapping_x - by calculating the mod of the corner coords - for all RTC and nlooks
+            if mod_x_prev is None:
+                mod_x_prev = geo_transformation_rtc_imagery[0] % geo_transformation_rtc_imagery[1]
+
+            elif abs(mod_x_prev - geo_transformation_rtc_imagery[0] % geo_transformation_rtc_imagery[1]) > maxerr_coord:
+                print(f'snapping_x anomaly detected from : {os.path.basename(path_rtc_imagery)}')
+                flag_rtn = True
+
+            elif abs(mod_x_prev - geo_transformation_nlooks[0] % geo_transformation_nlooks[1]) > maxerr_coord:
+                print(f'snapping_x anomaly detected from : {os.path.basename(path_nlooks)}')
+                flag_rtn = True
+
+            # snapping_y - by calculating the mod of the corner coords - for all RTC and nlooks
+            if mod_y_prev is None:
+                mod_y_prev = geo_transformation_rtc_imagery[3] % geo_transformation_rtc_imagery[5]
+            
+            elif abs(mod_y_prev - geo_transformation_rtc_imagery[3] % geo_transformation_rtc_imagery[5]) > maxerr_coord:
+                print(f'snapping_y anomaly detected from : {os.path.basename(path_rtc_imagery)}')
+                flag_rtn = True
+
+            elif abs(mod_y_prev - geo_transformation_nlooks[3] % geo_transformation_nlooks[5]) > maxerr_coord:
+                print(f'snapping_y anomaly detected from : {os.path.basename(path_nlooks)}')
+                flag_rtn = True
+
+            raster_rtc_imagery = None
+            raster_nlooks = None
+
+        # `list_rtc_imagery` and `list_nooks` are looking good so far.
+        # Now check `geogrid_mosaic` with the parameters from the tests so far
+
+        #Check spacing: Use spacing_x_prev, spacing_y_prev
+        if geogrid_mosaic is not None:
+            if spacing_x_prev != geogrid_mosaic.spacing_x:
+                print('spacing_x of the input rasters does not match with that of the mosaic geogrid')
+                flag_rtn = True
+
+            if spacing_y_prev != geogrid_mosaic.spacing_y:
+                print('spacing_y of the input rasters does not match with that of the mosaic geogrid')
+                flag_rtn = True
+
+            #Checo projection: Use str_proj_prev
+            srs_mosaic=osr.SpatialReference()
+            srs_mosaic.ImportFromEPSG(geogrid_mosaic.epsg)
+            if str_proj_prev != srs_mosaic.ExportToWkt():
+                print('Projection of the rasters do not match with that of the mosaic geogrid')
+                flag_rtn = True
+
+
+            #Check snapping: Use mod_x_prev, mod_y_prev
+            if abs(mod_y_prev - geogrid_mosaic.start_x % geogrid_mosaic.spacing_x) > maxerr_coord:
```

Comment:
I think you don't want to compare `mod_y_prev` from Y coordinate with start/spacing of X. `%` has precedence over `-`. Is this line really correct?

Reply:
Thanks for catching this. I've fixed this line.